File: python/armstrong-numbers/armstrong_numbers.py
from functools import reduce


def is_armstrong_number(number: int) -> bool:
    if number == 0:
        return True

    numbers = [int(x) for x in str(number)]

    soma = reduce(lambda y, _: pow(y, len(numbers)), numbers)

    return True if soma == number else False

File: typescript/armstrong-numbers/armstrong-numbers.ts
const splitNumber = (x: number): readonly number[] =>
  `${x}`.split('').map(Number);

const sum = (nums: readonly number[]): number =>
  nums.reduce((acc, num) => acc + Math.pow(num, nums.length), 0);

export const isArmstrongNumber = (digit: number): boolean =>
  sum(splitNumber(digit)) === digit ? true : false;
